core/src/test/java/hudson/util/AtomicFileWriterTest.java

```diff
@@ -40,12 +31,40 @@
 
 public class AtomicFileWriterTest {
     private static final String PREVIOUS = "previous value \n blah";
+    /**
+     * Provides access to the default permissions given to a new file. (i.e. indirect way to get umask settings).
+     * <p><strong>BEWARE: null on non posix systems</strong></p>
+     */
+    @Nullable
+    private static Set<PosixFilePermission> DEFAULT_GIVEN_PERMISSIONS;
     @Rule
     public TemporaryFolder tmp = new TemporaryFolder();
     File af;
     AtomicFileWriter afw;
     String expectedContent = "hello world";
 
+    @BeforeClass
+    public static void computePermissions() throws IOException {
+        final File tempDir = com.google.common.io.Files.createTempDir();
```

Comment:
why are you introducing this - you have `TemporaryFolder` which can [create directories](http://junit.org/junit4/javadoc/4.12/org/junit/rules/TemporaryFolder.html#newFolder(java.lang.String))...